elastic/kibana
# [Advanced settings] Add settings allowlist

Addresses https://github.com/elastic/kibana/issues/160411

## Summary

This PR adds functionality for filtering out advanced settings that are not relevant for serverless.

For context, we need to build an Advanced settings page in serverless which only contains a set of the existing settings. We will reuse the section registry (https://github.com/elastic/kibana/pull/163502) from the original Advanced settings plugin as well as its UI components which will also be extracted into a separate package. The app will be registered from inside the `serverless` plugin.

In order to only display the settings that are relevant for serverless, we need to make some changes to the uiSettings service. The implementation in this PR leverages the existing `readonly` uiSettings param and adds the `setAllowlist()` method which is called by the serverless plugin to set an allowlist of setting keys.

**Testing in serverless:**
1. Set `advanced_settings.enabled: true` to enable the Advanced settings app in serverless: https://github.com/elastic/kibana/blob/5b216c6ea94e739fea1f161f0bbce5a57ae44c02/config/serverless.yml#L53
2. Start Es with `yarn es serverless --ssl` and Kibana with `yarn serverless-{mode} --ssl` in any serverless mode.
3. Navigate to `app/management/kibana/settings`
4. Verify that the app only displays the settings from `packages/serverless/settings/common/index.ts` (these are the settings, relevant for all projects in serverless) as well as the settings from the corresponding project package `packages/serverless/settings/{mode}_project/index.ts`.
5. Verify that the app is functioning correctly.

**Testing in self-managed:**
1. Start Es with `yarn es snapshot` and Kibana with `yarn start`.
2. Go to Stack Management > Advanced settings
3. Verify that all settings are displayed as usual.
4. Verify that the app is functioning correctly.

If your team is a code owner of any of the serverless project plugins, please review the corresponding package `packages/serverless/settings/{search/observanility/security}_project/index.ts` where you've been added as an owner and test in the serverless solution accordingly.




### For maintainers
